packages/apps/reputation-oracle/server/src/common/utils/index.ts

```diff
@@ -0,0 +1,161 @@
+   * @param {StorageParams} clientParams - Configuration parameters for cloud storage
+   * @param {StorageCredentials} [credentials] - Optional. Cloud storage access data. If credentials are not provided, use anonymous access to the bucket
+   * @returns {Promise<UploadFile>} - Uploaded file with key/hash
+   */
+export async function copyFileFromURLToBucket(
+    url: string,
+    destBucket: string,
+    clientParams: StorageParams,
+    credentials?: StorageCredentials,
+): Promise<UploadFile> {
+    try {
+        const client: Minio.Client = new Minio.Client({
+            ...clientParams,
+            accessKey: credentials?.accessKey ?? '',
+            secretKey: credentials?.secretKey ?? '',
+        });
+
+        const { data } = await axios.get(url, { responseType: 'stream' });
+
+        const hash = await hashStream(data);
+        const key = `s3${hash}.zip`;
+
+        const stream = new PassThrough();
+        data.pipe(stream);
+
+        await new Promise((resolve, reject) => {
+            client.putObject(destBucket, key, stream, (err, etag) => {
+                if (err) reject(err);
+                else resolve(etag);
+            });
+        });
+
+        console.log(`File from ${url} copied to ${destBucket}/${key}`);
+
+        return {
+            key,
+            url: `${clientParams.useSSL ? 'https' : 'http'}://${
+              clientParams.endPoint
+            }${
+              clientParams.port ? `:${clientParams.port}` : ''
+            }/${destBucket}/${key}`,
+            hash,
+          };
+    } catch (error) {
+        console.error('Error copying file:', error);
+        throw new Error('File not uploaded');
+    }
+}
+
+function hashStream(stream: Readable): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const hash = crypto.createHash('sha1');
+        
+        stream.on('data', (chunk) => {
+            hash.update(chunk);
+        });
+
+        stream.on('end', () => {
+            resolve(hash.digest('hex'));
+        });
+
+        stream.on('error', (error) => {
+            reject(error);
+        });
+    });
+}
+
+
+type InputFile = {
+    type: 'path',
+    path: string,
+} | {
+    type: 'object',
+    content: any,
+};
+
+// TODO: Can be used for future implementaton in SDK Storage module
+/**
+   * **Upload file to cloud storage**
+   *
+   * @param {File[]} files - Files to upload
+   * @param {string} bucket - Bucket name
+   * @param {StorageParams} params - Cloud storage params
+   * @param {StorageCredentials} credentials - Optional. Cloud storage access data. If credentials is not provided - use an anonymous access to the bucket
+   * @returns {Promise<UploadFile>} - Uploaded file with key/hash
+   */
+export async function uploadFiles(
+    files: InputFile[],
+    bucket: string,
+    clientParams: StorageParams,
+    credentials?: StorageCredentials,
+): Promise<UploadFile[]> {
+    const client: Minio.Client = new Minio.Client({
```

Comment:
I think creating Minio client everytime we need to interact with cloud storage is not good from connection perspective. Doesn't it really help, if we have singleton of StorageClient, and then use that one to upload/download files? I think that way we can better manage the connection to storage cloud. I'm not entirely sure, though.

Reply:
This implantation was added as a temporary solution until SDK Storage is improved. I agree with you, it is better to have one connection somewhere at the global level when initializing the application.

Reply:
If creating the client is a problem, would passing a client instance be a solution?